insights/parsers/yumlog.py

```diff
@@ -28,124 +28,21 @@
 
 
-Entry = namedtuple('Entry', field_names='idx timestamp state pkg')
-"""namedtuple: Represents a line in ``/var/log/yum.log``."""
-
-
 @parser(Specs.yum_log)
-class YumLog(Parser):
-    """Class for parsing /var/log/yum.log"""
-
-    ERASED = 'Erased'
-    """Package Erased"""
-
-    INSTALLED = 'Installed'
-    """Package Installed"""
-
-    UPDATED = 'Updated'
-    """Package Updated"""
-
-    STATES = set([ERASED, INSTALLED, UPDATED])
-
-    def parse_content(self, content):
-        """Parses contents of each line in ``/var/log/yum.log``.
-
-        Each line in the file contains 5 fields that are parsed into the
-        attributes of ``Entry`` instances:
-
-            - month
-            - day
-            - time
-            - state
-            - package
-
-        The month, day, and time form the ``Entry.timestamp``.
-        ``Entry.state`` contains the state of the package, one of ``ERASED``,
-        ``INSTALLED``, or ``UPDATED``. ``Entry.pkg`` contains the ``InstalledRpm``
-        instance corresponding to the parse package. ``Entry.idx`` is the
-        zero-based line number of the ``Entry`` in the file.  It can be
-        used to tell ordering of events.
-
-        Parameters:
-            content (list): Lines of ``/var/log/yum.log`` to be parsed.
-
-        Raises:
-            ParseException: if a line can't be parsed for any reason.
-        """
-        self.data = []
-        self.pkgs = defaultdict(list)
-        for idx, line in enumerate(get_active_lines(content)):
-            if not any(s in line for s in self.STATES):
-                continue
-            try:
-                line = line.replace(': 100', '')
-                month, day, time, state, pkg = line.split()[:5]
-                timestamp = ' '.join([month, day, time])
-                state = state.rstrip(':')
-                pkg = pkg.split(':')[-1].strip()
-                if state == self.ERASED:
-                    pkg = InstalledRpm({'name': pkg})
-                else:
-                    pkg = InstalledRpm.from_package(pkg)
-                e = Entry(idx, timestamp, state, pkg)
-                self.data.append(e)
-                self.pkgs[pkg.name].append(e)
-            except:
-                raise ParseException('YumLog could not parse', line)
-
-        self.pkgs = dict(self.pkgs)
-
-    def __len__(self):
-        return len(self.data)
-
-    def __iter__(self):
-        for d in self.data:
-            yield d
-
-    def _packages_currently(self, states):
-        result = {}
-        for k, v in self.pkgs.items():
-            e = v[-1]
-            if e.state in states:
-                result[k] = e
-
-        return result
-
-    @property
-    def present_packages(self):
-        """``list`` of latest ``Entry`` instances for installed packages."""
-        return self._packages_currently([self.INSTALLED, self.UPDATED])
+class YumLog(LogFileOutput):
+    """Class for parsing ``/var/log/yum.log`` file."""
+    pass
```

Comment:
Like the `Syslog`, could you re-implement a `_parse_line` to parse each line to different parts, e.g. timestamp, state, package (in type of InstalledRpm)?

https://github.com/RedHatInsights/insights-core/blob/869bbb4c745fd6559a7276dc88a4400a55fbcde3/insights/core/__init__.py#L1296

Reply:
Added